index.js

```diff
@@ -1,33 +1,59 @@
 const fetch = require('node-fetch'); // Automatically excluded in browser bundles
 
-async function api(endpoint, token) {
-	token = token ? `&access_token=${token}` : '';
-	const response = await fetch(`https://api.github.com/repos/${endpoint}${token}`);
+// Matches '/<re/po>/tree/<ref>/<dir>'
+const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
+
+function parseResource(res) {
+	if (typeof res === 'string') {
+		try {
+			const parsedUrl = new URL(res);
+			res = {api: 'https://api.github.com'};
+			[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname);
+			if (parsedUrl.hostname !== 'github.com') {
+				res.api = `${parsedUrl.protocol}//${parsedUrl.host}/api/v3`;
+			}
+		} catch {
+			throw new Error('Unable to parse GitHub URL');
+		}
+	}
+
+	// TODO: Validate
+	return res;
+}
+
+async function githubApi(api, endpoint, token) {
+	const response = await fetch(`${api}/repos/${endpoint}`, {
+		headers: {
+			Authorization: `Bearer ${token}`
+		}
+	});
 	return response.json();
 }
 
 // Great for downloads with few sub directories on big repos
 // Cons: many requests if the repo has a lot of nested dirs
+
+// api = 'https://api.github.com',
+// user,
+// repository,
+// ref = 'HEAD',
+// directory,
+
 async function viaContentsApi({
-	user,
-	repository,
-	ref = 'HEAD',
-	directory,
+	resource,
 	token,
 	getFullData = false
 }) {
+	const res = parseResource(resource);
```

Comment:
I think you can reuse it:
```suggestion
	resource = parseResource(resource);
```
Perhaps you can actually only parse the string:

```suggestion
	if (typeof resource === 'string') {
		resource = parseResource(resource);
	}
```